### PR TITLE
[SPARK-14512][DOC] Add python example for QuantileDiscretizer

### PR DESCRIPTION
**Pull request by zhengruifeng**
## What changes were proposed in this pull request?

Add the missing python example for QuantileDiscretizer
## How was this patch tested?

manual tests

### CODE REVIEW

**Comment by holdenk** [comment]
Looks good to me, confirmed the example runs as expected.

**Comment by holdenk** [comment]
cc @davies ?

File: examples/src/main/python/ml/quantile_discretizer_example.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+from __future__ import print_function
+
+from pyspark import SparkContext
+from pyspark.sql import SQLContext
+# $example on$
+from pyspark.ml.feature import QuantileDiscretizer
+# $example off$
+
+if __name__ == "__main__":
+    sc = SparkContext(appName="QuantileDiscretizerExample")
```
Comment (MLnick) [inline examples/src/main/python/ml/quantile_discretizer_example.py]: Examples have been updated to use SparkSession, see e.g. https://github.com/apache/spark/blob/master/examples/src/main/python/ml/binarizer_example.py. 

Also, there is a PR to indent the builder pattern #12911 so you should follow that style (also refer #12809)

**Comment by MLnick** [comment]
@zhengruifeng sorry for the delay. The core of the example looks good, it just now needs to be updated to use `SparkSession` as per my comment. Thanks!

**Comment by zhengruifeng** [comment]
@MLnick  Updated. Thanks for review this.

**Comment by yanboliang** [comment]
LGTM. Thanks!

**Comment by davies** [comment]
Merging this into master and 2.0, thanks!

**Pull request closed without merging**